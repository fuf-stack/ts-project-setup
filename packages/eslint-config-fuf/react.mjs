import { configs, rules } from 'eslint-config-airbnb-extended';
import jsxA11yPlugin from 'eslint-plugin-jsx-a11y';
import reactPlugin from 'eslint-plugin-react';
import reactHooksPlugin from 'eslint-plugin-react-hooks';

import baseConfig from './base.mjs';
import fufStorybook from './fuf/fuf-react-storybook.mjs';
import fufReact from './fuf/fuf-react.mjs';

export default [
  // Base Config
  ...baseConfig,

  // Register React-related plugins explicitly for flat config resolution
  {
    name: 'fuf/react/plugins',
    plugins: {
      react: reactPlugin,
      'react-hooks': reactHooksPlugin,
      'jsx-a11y': jsxA11yPlugin,
    },
  },

  // Airbnb React Recommended Config
  ...configs.react.recommended,
  // Airbnb React TypeScript Config
  ...configs.react.typescript,
  // Strict React Config
  rules.react.strict,

  // FUF React Configs
  fufReact,
  fufStorybook,
];
