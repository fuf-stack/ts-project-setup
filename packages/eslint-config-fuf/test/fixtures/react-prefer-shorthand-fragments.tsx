/* eslint-disable */

/* eslint-disable import-x/no-extraneous-dependencies */

import { Fragment } from 'react';

// prefer shorthand fragments
const Hello = () => <Fragment>test</Fragment>;

export default Hello;
