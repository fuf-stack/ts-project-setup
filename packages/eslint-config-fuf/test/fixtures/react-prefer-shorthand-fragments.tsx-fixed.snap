/* eslint-disable import-x/no-extraneous-dependencies */

import { Fragment } from 'react';

// prefer shorthand fragments
const Hello = () => <>test</>;

export default Hello;
